packages/anoncreds-rs/src/services/AnonCredsRsHolderService.ts

```diff
@@ -0,0 +1,380 @@
+    // the credential records with some properties of the schema it is based on. This means that Schema should have been
+    // previously retrieved and present in the storage. Can we assume that at this point it will be stored?
+    const schemaRecord = await agentContext.dependencyManager
+      .resolve(AnonCredsSchemaRepository)
+      .getBySchemaId(agentContext, options.credential.schema_id)
+
+    const revocationRegistryDefinition = options.revocationRegistry?.definition
+      ? RevocationRegistryDefinition.load(JSON.stringify(options.revocationRegistry.definition))
+      : undefined
+
+    const credentialId = options.credentialId ?? uuid()
+
+    const processedCredential = Credential.load(JSON.stringify(options.credential)).process({
+      credentialDefinition: CredentialDefinition.load(JSON.stringify(options.credentialDefinition)),
+      credentialRequestMetadata: CredentialRequestMetadata.load(JSON.stringify(options.credentialRequestMetadata)),
+      masterSecret: MasterSecret.load(JSON.stringify({ value: { ms: linkSecretRecord.value } })),
+      revocationRegistryDefinition,
+    })
+
+    const credentialRepository = agentContext.dependencyManager.resolve(AnonCredsCredentialRepository)
+
+    await credentialRepository.save(
+      agentContext,
+      new AnonCredsCredentialRecord({
+        credential: JSON.parse(processedCredential.toJson()) as AnonCredsCredential,
+        credentialId,
+        linkSecretId: linkSecretRecord.linkSecretId,
+        issuerId: options.credentialDefinition.issuerId,
+        schemaName: schemaRecord.schema.name,
+        schemaIssuerId: schemaRecord.schema.issuerId,
+        schemaVersion: schemaRecord.schema.version,
+      })
+    )
+
+    return credentialId
+  }
+
+  public async getCredential(
+    agentContext: AgentContext,
+    options: GetCredentialOptions
+  ): Promise<AnonCredsCredentialInfo> {
+    const credentialRecord = await agentContext.dependencyManager
+      .resolve(AnonCredsCredentialRepository)
+      .getByCredentialId(agentContext, options.credentialId)
+
+    const attributes: { [key: string]: string } = {}
+    for (const attribute in credentialRecord.credential.values) {
+      attributes[attribute] = credentialRecord.credential.values[attribute].raw
+    }
+    return {
+      attributes,
+      credentialDefinitionId: credentialRecord.credential.cred_def_id,
+      credentialId: credentialRecord.credentialId,
+      schemaId: credentialRecord.credential.schema_id,
+      credentialRevocationId: credentialRecord.credentialRevocationId,
+      revocationRegistryId: credentialRecord.credential.rev_reg_id,
+    }
+  }
+
+  public async deleteCredential(agentContext: AgentContext, credentialId: string): Promise<void> {
+    const credentialRepository = agentContext.dependencyManager.resolve(AnonCredsCredentialRepository)
+    const credentialRecord = await credentialRepository.getByCredentialId(agentContext, credentialId)
```

Comment:
We also have `deleteById` now I think. Or do you do this so an error is thrown if it doesn't exist?

Reply:
This is because I'm not using AnonCredsCredentialRecord.id but a separate AnonCredsCredentialRecord.credentialId. However, I don't see the benefit on having a dedicated property for credentialId in this case, so I'm thinking on simplifying it and just use the id. What do you think?

Reply:
I think it _could_ work. However it has caused issues for me in the past. E.g with using the `did` as an id for a record 🤦 

But as we always generate an uuid I think we should be fine here?

Reply:
Yeah exactly, we are always doing so internally, either at xxxFormatService or xxxHolderService.

Reply:
But still, based on previous experiences maybe it will be better to keep it as is and conceptually separate the storage record Id from other queries.